addon/components/infinity-loader.js

```diff
@@ -1,106 +1,65 @@
 import Ember from 'ember';
 import InfinityPromiseArray from 'ember-infinity/lib/infinity-promise-array';
+import InViewportMixin from 'ember-in-viewport';
+import { run } from '@ember/runloop';
+import { computed } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import Component from '@ember/component';
 
-const InfinityLoaderComponent = Ember.Component.extend({
+const InfinityLoaderComponent = Component.extend(InViewportMixin, {
   classNames: ["infinity-loader"],
-  classNameBindings: ["infinityModelContent.reachedInfinity"],
+  classNameBindings: ["infinityModelContent.reachedInfinity", "viewportEntered:in-viewport"],
   guid: null,
   eventDebounce: 10,
   loadMoreAction: 'infinityLoad',
   loadingText: 'Loading Infinite Model...',
   loadedText: 'Infinite Model Entirely Loaded.',
-  destroyOnInfinity: false,
```

Comment:
Is there a reason we can't do `destroyOnInfinity` anymore?  It's kinda a cool feature for anyone that doesn't want a "Loaded all Records." state on their loader

Reply:
Ahep that is my mistake!